fendermint/testing/materializer/src/manifest.rs

```diff
@@ -0,0 +1,223 @@
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Account {}
+
+/// Account balance.
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Balance(#[serde_as(as = "IsHumanReadable")] pub TokenAmount);
+
+/// Ways we can hook up with IPC contracts on the rootnet.
+///
+/// The rootnet is generally expected to be Calibration net,
+/// where IPC contracts are deployed from Hardhat, and multiple
+/// instances of the gateway exist, each with a different version
+/// and an address we learn after deployment.
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum IpcDeployment {
+    /// Use one of the existing deployments, given by the delegated address of
+    /// the Gateway and Registry contracts.
+    Existing {
+        #[serde_as(as = "IsHumanReadable")]
+        gateway: Address,
+        #[serde_as(as = "IsHumanReadable")]
+        registry: Address,
+    },
+    /// Deploy a new IPC contract stack using one of the accounts.
+    /// This can take a long time, but ensures we are testing with
+    /// contracts that have the same version as the client.
+    New { deployer: AccountId },
+}
+
+/// The rootnet, ie. the L1 chain, can already exist and be outside our control
+/// if we are deploying to Calibration, or it might be a chain we provision
+/// with CometBFT and Fendermint.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum Rootnet {
+    /// Existing L1 running outside our control.
+    ///
+    /// This implies using some sort of Faucet to get balances for the accounts.
+    External {
+        /// Indicate whether we have to (re)deploy the IPC contract or we can use an existing one.
+        deployment: IpcDeployment,
+    },
+
+    /// Provision a new chain to run the L1.
+    ///
+    /// It is assumed that a newly provisioned chain will have built-in support for IPC,
+    /// e.g. the way Fendermint deploys IPC actors at well-known addresses.
+    New {
+        /// Collateral of the initial validator set.
+        validators: CollateralMap,
+        /// Balances of the accounts in the rootnet.
+        ///
+        /// These balances will go in the genesis file.
+        balances: BalanceMap,
+        /// Nodes that participate in running the root chain.
+        nodes: NodeMap,
+    },
+}
+
+/// An IPC subnet.
+///
+/// The balance of the account on the parent subnet, as declared in this manifest,
+/// _does not_ have to account for the collateral/balance we have to take from it to join/fund the subnet.
+/// When we create the testnet configuration from the manifest we will account for this with a rollup,
+/// so we don't have to do that much mental arithmetic and run into frustrating errors during setup.
+/// If we want to test trying to join with more than what we have, we can do so in the integration test.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Subnet {
```

Comment:
I think it's fine to go without bootstrappers for now, assuming we're bootstrapping in daisy chain, but we may want to add a TODO for a bootstrap strategy? Or, we could add it here and only allow daisy chaining as an option for now.